public/app/features/explore/state/query.ts

```diff
@@ -696,6 +698,114 @@ export const runQueries = createAsyncThunk<void, RunQueriesOptions>(
   }
 );
 
+interface RunLoadMoreLogsQueriesOptions {
+  exploreId: string;
+  absoluteRange: AbsoluteTimeRange;
+}
+/**
+ * Supplementary action to run queries that request more results, and dispatches sub-actions based
+ * on which result viewers are active
+ */
+export const runLoadMoreLogsQueries = createAsyncThunk<void, RunLoadMoreLogsQueriesOptions>(
+  'explore/runQueries',
+  async ({ exploreId, absoluteRange }, { dispatch, getState }) => {
+    dispatch(cancelQueries(exploreId));
+
+    const correlations$ = getCorrelations(exploreId);
+    const { datasourceInstance, containerWidth, queryResponse, correlationEditorHelperData } =
+      getState().explore.panes[exploreId]!;
+
+    const isCorrelationEditorMode = getState().explore.correlationEditorDetails?.editorMode || false;
+    const isLeftPane = Object.keys(getState().explore.panes)[0] === exploreId;
+    const showCorrelationEditorLinks = isCorrelationEditorMode && isLeftPane;
+    const defaultCorrelationEditorDatasource = showCorrelationEditorLinks ? await getDataSourceSrv().get() : undefined;
+    const interpolateCorrelationHelperVars =
+      isCorrelationEditorMode && !isLeftPane && correlationEditorHelperData !== undefined;
+    let newQuerySource: Observable<ExplorePanelData>;
```

Comment:
Maybe it could be extracted to a helper function and reused in runQueries? (+ scopedVars calculations which happens a few lines below). Something like:

```typescript

async function getCorrelationsInfo(
  getState: () => StoreState,
  exploreId: string,
  correlationEditorHelperData: ExploreCorrelationHelperData | undefined
) {
  const isCorrelationEditorMode = getState().explore.correlationEditorDetails?.editorMode || false;
  const isLeftPane = Object.keys(getState().explore.panes)[0] === exploreId;
  const showCorrelationEditorLinks = isCorrelationEditorMode && isLeftPane;
  const defaultCorrelationEditorDatasource = showCorrelationEditorLinks ? await getDataSourceSrv().get() : undefined;
  const interpolateCorrelationHelperVars =
    isCorrelationEditorMode && !isLeftPane && correlationEditorHelperData !== undefined;
  let scopedVars: ScopedVars = {};
  if (interpolateCorrelationHelperVars && correlationEditorHelperData !== undefined) {
    Object.entries(correlationEditorHelperData?.vars).forEach((variable) => {
      scopedVars[variable[0]] = { value: variable[1] };
    });
  }
  return { showCorrelationEditorLinks, defaultCorrelationEditorDatasource, scopedVars };
}

// ---- and then:

const { showCorrelationEditorLinks, defaultCorrelationEditorDatasource, scopedVars } =
    await getCorrelationsInfo(getState, exploreId, correlationEditorHelperData);
```

Reply:
Great suggestion! Added a new function to hold that code for reuse and placed it in the utils module.